userspace/chisel/chisel.cpp

```diff
@@ -1170,33 +1170,42 @@ bool sinsp_chisel::openfile(string filename, OUT ifstream* is)
 	return false;
 }
 
-void sinsp_chisel::load(string cmdstr)
+void sinsp_chisel::load(string cmdstr, bool is_file)
 {
-	m_filename = cmdstr;
-	trim(cmdstr);
+	if (is_file) {
+		m_filename = cmdstr;
+		trim(cmdstr);
+	} else {
+		m_filename = "<in-memory-string>";
+	}
 
 	ifstream is;
+	std::string scriptstr;
 
-	//
-	// Try to open the file with lua extension
-	//
-	if(!openfile(m_filename + ".lua", &is))
-	{
+	if (is_file) {
 		//
-		// Try to open the file as is
+		// Try to open the file with lua extension
 		//
-		if(!openfile(m_filename, &is))
+		if(!openfile(m_filename + ".lua", &is))
 		{
-			throw sinsp_exception("can't open file " + m_filename);
+			//
+			// Try to open the file as is
+			//
+			if(!openfile(m_filename, &is))
+			{
+				throw sinsp_exception("can't open file " + m_filename);
+			}
 		}
-	}
 
+		//
+		// Load the file
+		//
+		std::istreambuf_iterator<char> eos;
+		std::string scriptstr(std::istreambuf_iterator<char>(is), eos);
```

Comment:
Ouch you are not actually using afore-declared `scriptstr` here :)

Reply:
You are right, would something like this work/be acceptable?
```c++
std::stringstream scriptstream;
scriptstream << is;
scriptstr = scriptstream.str();
```

Reply:
Wouldn't this work?
`scriptstr.assign(std::istreambuf_iterator<char>(is), eos);`

Otherwise yours looks fine too!

Reply:
I like yours better, I'm not 100% versed in the ways of C++ yet, so it's nice to see this kind of thing, thanks!

Reply:
If that actually works, ofc :laughing: